### PR TITLE
cargo: workspace: use package table

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview
Use the package table feature from cargo to make it easier to update editions and versions for all of our crates.

https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table

I skipped the libraries directory as we probably want to keep those standalone.

### Testing Strategy

travis

### TODO or Help Wanted

This is blocked on updating the rust nightly.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -93,6 +93,11 @@ exclude = [
     "tools/usb/control-test",
 ]
 
+[workspace.package]
+version.workspace = true
+authors.workspace = true
+edition.workspace = true
```
Comment (hudson-ayers) [inline Cargo.toml:99]:
```suggestion
version.workspace = "2.1.0"
authors.workspace = ["Tock Project Developers <tock-dev@googlegroups.com>"]
edition.workspace = "2021"
```

That is my understanding of how this should look based on https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table . Not sure what version we should put there

**Review by hudson-ayers — CHANGES_REQUESTED** [review]
`tools/` could probably share the authors field, but not the version / edition

File: Cargo.toml (hunk 1 above)
Comment (brghena) [reply Cargo.toml:99]: I was about to ask about this. It seems like _some_ file needed to actually have contents.

Comment (bradjc) [reply Cargo.toml]: Ooops my find and replace was too good

File: Cargo.toml [hunk 2]
```diff
@@ -93,6 +93,11 @@ exclude = [
     "tools/usb/control-test",
 ]
 
+[workspace.package]
+version.workspace = "0.1.0"
```
[the remark this answers is not in the document]
Comment (hudson-ayers) [reply Cargo.toml:97]: sounds good

**Review by hudson-ayers — APPROVED** [review]

**Review by alistair23 — DISMISSED** [review]

**Review by hudson-ayers — APPROVED** [review]

**Review by hudson-ayers — DISMISSED** [review]

**Review by alistair23 — DISMISSED** [review]

**Review by lschuermann — APPROVED** [review]
Looks good! I assume we'd still wait for #3303 to land?

**Comment by bradjc** [comment]
> Looks good! I assume we'd still wait for #3303 to land?

We don't need it for this PR.

**Comment by alexandruradovici** [comment]
PR #3300 has all the changes from here, we just need to bring those to master.

**Review by lschuermann — APPROVED** [review]

**Review by hudson-ayers — APPROVED** [review]
bors r+

**Review by lschuermann — APPROVED** [review]
For the record, these boards had duplicate `build` keys which I've removed:
- `boards/apollo3/lora_things_plus/Cargo.toml`
- `boards/apollo3/redboard_artemis_nano/Cargo.toml`

bors r+

**Pull request merged**